board.cpp

```diff
@@ -0,0 +1,190 @@
+#include "board.h"
+
+const std::vector<CreateFn> Board::pieceNames = { &Create<Square>, &Create<TBlock>, &Create<LBlockL>, &Create<LBlockR>, &Create<Straight>, &Create<ZBlock>, &Create<SBlock>};
+
+Board::Board() {
+	this->board = std::vector<int>(BOARD::BOARD_SIZE, 0);
+	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
+	this->currentHeight = 0;
+	this->currentPiece = new Piece();
+	this->absPiecePositionCol = 0;
+	this->absPiecePositionRow = 0;
+	this->rotation = 0;
+}
+
+int& Board::operator[] (int rowCol){
+	return board[rowCol];
+}
+
+
+int& Board::operator[] (indices rowCol) {
+	return board[rowCol.i * 10 + rowCol.j];
+}
+
+inline int Board::getPieceCol(int i) {
+	return this->absPiecePositionCol + (*(this->currentPiece->relXPositions))[rotation * 4 + i];
```

Comment:
I dont know what 4 is? Also based on the naming of this function I would expect you to just return this->absPiecePositionCol. I am uncertain of what the other part of this expression indicates on first glance.